### PR TITLE
Support public read access ACL

### PR DESCRIPTION
**Pull request by ben181231**
connect oursky/skygear-server#647

### CODE REVIEW

**Comment by cheungpat** [comment]
I looked through it and it lgtm but I wonder if @rickmak can have a second look because he knows  better about the ACL

**Comment by ben181231** [comment]
@cheungpat Okay. It should be tested together with the updates on SDKs.

File: skydb/pq/builder.go [hunk 1]
```diff
@@ -262,19 +262,17 @@ func (p accessPredicateSqlizer) ToSql() (sql string, args []interface{}, err err
...
+		b.WriteString(fmt.Sprintf(`_access @> '[{"role": %s}]' OR `, escapedRole))
 	}
-	b.WriteString(`_access @> '[{"user_id":`)
-	escapedID, _ := json.Marshal(p.user.ID)
-	if err != nil {
-		panic("unexpected serialize error on user_id")
-	}
-	b.Write(escapedID)
-	b.WriteString(`}]' OR _access IS NULL OR _owner_id = ?)`)
+	b.WriteString(fmt.Sprintf(`_access @> '[{"user_id": "%s"}]' OR `, p.user.ID))
```
Comment (rickmak) [inline skydb/pq/builder.go]: `json.Marshal` is used to escaped the `user_id`

If you use fmt.Sprintf and user id is string like `oursky"rickmak"`, the SQL will not able to execute.

**Comment by rickmak** [comment]
Other looks fine to me.

File: skydb/pq/builder.go (hunk 1 above)
Comment (ben181231) [inline skydb/pq/builder.go]: Oh. I see.

**Pull request merged**